finalProject-2020-q3/code/BoardMovements.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace finalProject_2020_q3.code
+{
+    class BoardMovements
+    {
+        public static CellList AllCrossCells(Cell[,] piecesOnBoard, Cell position)
+        {
+            CellList left = GetCellsLeft(piecesOnBoard, position, new CellList());
+            CellList right = GetCellsRight(piecesOnBoard, position, left);
+            CellList down = GetCellsDown(piecesOnBoard, position, right);
+            CellList response = GetCellsUp(piecesOnBoard, position, down);
+            return response;
+        }
+
+        public static CellList AllCellsDiagonal(Cell[,] piecesOnBoard, Cell position)
+        {
+            CellList downLeft = GetCellDownLeft(piecesOnBoard, position, new CellList());
+            CellList downRight = GetCellDownRight(piecesOnBoard, position, downLeft);
+            CellList upLeaft = GetCellUpLeaft(piecesOnBoard, position, downRight);
+            CellList response = GetCellUpRight(piecesOnBoard, position, upLeaft);
+            return response;
+        }
+
+        private static CellList GetCellUpLeaft(Cell[,] piecesOnBoard, Cell cell, CellList response)
+        {
+            Cell next = CellMovements.UpLeftDiagonal(piecesOnBoard, cell);
+            if (ValidatCell(next))
+            {
+                response.Add(next);
+                return next.IsEmpty() ?
+                    GetCellUpLeaft(piecesOnBoard, next, response) : response;
+            }
+            return response;
+        }
+
+        private static CellList GetCellUpRight(Cell[,] piecesOnBoard, Cell cell, CellList response)
+        {
+            Cell next = CellMovements.UpRightDiagonal(piecesOnBoard, cell);
+            if (ValidatCell(next))
+            {
+                response.Add(next);
+                return next.IsEmpty() ?
+                    GetCellUpRight(piecesOnBoard, next, response) : response;
+            }
+            return response;
+        }
+
+        private static CellList GetCellDownRight(Cell[,] piecesOnBoard, Cell cell, CellList response)
```

Comment:
@lufercc is it necessary to pass the response as parameter, maybe we could use the 'out' keyword if necessary